projects/core/src/state/event/state-event.service.ts

```diff
@@ -0,0 +1,73 @@
+import { Injectable, Type } from '@angular/core';
+import { ofType } from '@ngrx/effects';
+import { ActionsSubject } from '@ngrx/store';
+import { Observable } from 'rxjs';
+import { map } from 'rxjs/operators';
+import { EventService } from '../../event/event.service';
+import { ActionToEventMapping } from './action-to-event-mapping';
+
+/**
+ * Registers streams of ngrx actions as events source streams
+ */
+@Injectable({
+  providedIn: 'root',
+})
+export class StateEventService {
+  constructor(
+    protected actionsSubject: ActionsSubject,
+    protected eventService: EventService
+  ) {}
+
+  /**
+   * Registers an event source stream of specific events
+   * mapped from a given action type.
+   *
+   * This method can be chained. For example:
+   * ```
+   * service
+   *  .register({ event: EventA, action: ActionA })
+   *  .register({ event: EventB, action: ActionB });
+   * ```
+   *
+   * @param mapping mapping from action to event
+   *
+   * @returns the service
+   */
+  register<T>(mapping: ActionToEventMapping<T>): StateEventService {
+    this.eventService.register(mapping.event, this.getFromAction(mapping));
+    return this;
```

Comment:
Not sure if we gain much with this chaining solution here. 
It's not consistent with `eventService.register`, could be considered more confusing + it requires additional knowledge/docs.